tests/src/unknown_fields.rs

```diff
@@ -0,0 +1,76 @@
+use prost::{Message, UnknownField};
+
+mod unknown_fields {
+    include!(concat!(env!("OUT_DIR"), "/unknown_fields.rs"));
+}
+
+#[test]
+fn test_access_unknown_field() {
+    let message = unknown_fields::MessageWithExtraFields {
+        normal_field: "normal".to_string(),
+        extra_field: "extra".to_string(),
+    };
+    let mut encoded = Vec::new();
+    message.encode(&mut encoded).unwrap();
+    let message = unknown_fields::MessageWithUnknownFields::decode(&encoded[..])
+        .expect("Could not decode as MessageWithUnknownFields");
+
+    assert_eq!(
+        message.protobuf_unknown_fields,
+        vec![UnknownField {
+            tag: 2,
+            value: b"extra".to_vec()
+        }]
+    );
+}
+
+#[test]
+fn test_serialize_unknown_field() {
+    let message = unknown_fields::MessageWithUnknownFields {
+        normal_field: "normal".to_string(),
+        protobuf_unknown_fields: vec![UnknownField {
```

Comment:
Could you add a test in which the protobuf message has an actual field called `protobuf_unknown_fields`, of whatever type? I think we should clarify what behaviour we expect in that case, which I guess will boil down to one of the following:

- fail early with a friendly error message
- somehow try to escape the field name so that it does not conflict with this; I expect there must be a list of identifiers to avoid (e.g. Rust keywords) that are already escaped by prost -- or does it just ignore them?
- give this field a name that cannot possibly clash with any generated field, e.g. `_protobuf_unknown_fields` or similar.

Reply:
I did a quick test, and it prost removes leading underscores in protobuf field names, so I think we could name the field `_unknown_fields`?

Reply:
As a sidenote, if you define a message like this:
```
message Test {
  optional string _weird_field = 1;
  optional string weird_field = 2;
}
```

Prost fails with:
```
4 |     pub weird_field: ::std::option::Option<std::string::String>,
  |     ----------------------------------------------------------- `weird_field` first declared here
5 |     #[prost(string, optional, tag="2")]
6 |     pub weird_field: ::std::option::Option<std::string::String>,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ field already declared
```